src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -65,87 +66,81 @@
+        withNonSGAErrors.mapToPair(entry -> new Tuple2<>(entry._1(), entry._2().nonSGAStepsErrorMessages))
                 .saveAsObjectFile(outputDir);
     }
 
     /**
-     * Final return type of the whole process of SGA local assembly.
-     * assembledContigFile is the file containing the assembled contigs, if the process executed successfully, or null if not.
-     * runtimeInformation contains the runtime information logged along the process up until the process erred, if errors happen,
-     *  or until the last step if no errors occur along the line.
-     *  The list is organized along the process of executing the assembly pipeline, inside each entry of the list is
-     *      the name of the module,
-     *      return status of the module,
-     *      stdout message of the module, if any,
-     *      stderr message of the module, if any
+     * Converts from a line of text, delimited with a tab, where the first entry is the breakpoint ID and the second entry
+     *   is URI to its associated FASTQ file on HDFS.
+     * @param recordLine    a line of text containing the breakpoint ID and URI to associated FASTQ file
+     * @return              a pair constructed by splitting the string into the ID part and the URI part
+     *                      the URI part will be null if cannot be parsed
      */
     @VisibleForTesting
-    static final class SGAAssemblyResult implements Serializable{
-        private static final long serialVersionUID = 1L;
-
-        public final ContigsCollection assembledContigs;
-        public final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> runtimeInformation;
-
-        public SGAAssemblyResult(final File contigFile, final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> collectedRuntimeInfo){
-            ContigsCollection proxy = null;
-            try{
-                proxy = new ContigsCollection(contigFile);
-            } catch (final IOException e) { // on IO error caused by converting contigs FASTA file to memory, set in-memory struct to null and append error message
-                collectedRuntimeInfo.add(new Tuple2<>(ContigsCollection.class.getName(),
-                                                      new Tuple3<>(ExternalCommandlineProgramModule.ReturnStatus.STDIOFAIL, "", e.getMessage())));
-            }
-            assembledContigs = proxy;
-            runtimeInformation = collectedRuntimeInfo;
+    static Tuple2<Long, URI> assignFASTQToBreakpoints(final String recordLine){
+        final String[] recordForOneBreakPoint = recordLine.split("\t");
+        final Long breakpointID = Long.valueOf(recordForOneBreakPoint[0]);
+        try{
+            final URI fastqURI = new URI(recordForOneBreakPoint[1]);
+            return new Tuple2<>(breakpointID, fastqURI);
+        }catch (final URISyntaxException e){
```

Comment:
I don't think we should catch and swallow this exception here. We should probably re-throw it as a UserException (since it indicates that the input data was bad, containing improperly formed URI strings).